File: Assets/Scripts/Module/Invoke/FActions/SyncAction.cs
﻿using System.Collections;
using System.Collections.Generic;
using UnityEngine;
namespace FAction
{
    // 并发执行组合节点
    public class SyncAction : GroupAction
    {
        public override ActionType ActionType => ActionType.Sync;
        public override void ClearGroupInfo() { }

        public override void Replay()
        {
            base.Replay();
            foreach (var child in childs)
            {
                child.Replay();
            }
        }

        public override void Tick()
        {
            base.Tick();
            var finishCount = 0;
            for (int i = 0; i < ChildCount; i++)
            {
                var child = childs[i];
                if (!child.IsFinish)
                {
                    child.Tick();
                }

                if (child.IsFinish)
                {
                    finishCount++;
                }
            }

            if (finishCount == ChildCount)
            {
                this.IsFinish = true;
            }
        }
    }
}

File: Assets/.lpproj/Apis/Unity/CS/FAction/SyncAction.lua
---@class CS.FAction.SyncAction : CS.FAction.GroupAction
CS.FAction.SyncAction = {}

---@property readonly CS.FAction.SyncAction.ActionType : CS.FAction.ActionType
CS.FAction.SyncAction.ActionType = nil

---@return CS.FAction.SyncAction
function CS.FAction.SyncAction()
end

function CS.FAction.SyncAction:ClearGroupInfo()
end

function CS.FAction.SyncAction:Replay()
end

function CS.FAction.SyncAction:Tick()
end